ir/src/lib.rs

```diff
@@ -59,51 +53,26 @@ impl HarvestIR {
         self.representations.contains_key(&id)
     }
 
+    /// Returns all contained Representations of the given type.
+    pub fn get_by_representation<R: Representation>(&self) -> impl Iterator<Item = &R> {
+        // TODO: Add a `TypeId -> Id` map to HarvestIR that allows us to look these up without
+        // scanning through all the other representations.
+        self.representations
+            .values()
+            .filter_map(|repr| <dyn Any>::downcast_ref(repr))
```

Comment:
I suspect there is a problem here in the difference between the type itself and the type of a reference to the type, causing the `downcast_ref` to fail.

Reply:
Yeah that's probably it. I considered writing a unit test, but that's annoying because you need to create *multiple* test representations, so I thought "it's only three lines long, I'll skip".

Building that unit test now. Even if that's not the cause, it's worth doing.